prototype_v1/Student-View/modules/grade.R

```diff
@@ -0,0 +1,179 @@
+homework_UI <- function(id) {
+  tabPanel(title = "Homework"
+           , fluidRow(
+             column(width = 12
+                   , box(width = 12, status = "primary"
+                      , rHandsontableOutput(NS(id,"homework_grades"))
+                   )
+             )
+             , column(width = 12
+              , box(width = 12, status = "primary"
+                , echarts4rOutput(NS(id,"homework_grade_bar"), height = "300px")
+              )
+             )
+           )
+  )
+}
+
+homework_server <- function(id, r){
+  moduleServer(id, function(input, output, session){
+    df_filtered_homework_grades <- reactive({
+      req(r$is$auth)
+      student_name <- r$df_student %>%
+        filter(student_id == r$auth_student_id()) %>%
+        select(name) %>%
+        pull()
+      
+      homework_grade <- r$df_homework_table %>%
+        filter(`Student Name` == student_name)
+    })
+    
+    output$homework_grade_bar <- renderEcharts4r({
+      df <- df_filtered_homework_grades() %>%
+        select(-c(`Student Name`))
+      
+      df <- as_tibble(cbind(Homework = names(df), t(df))) %>%
+        mutate(grade = as.numeric(V2)) %>%
+        mutate(grade = grade / 100)
+      
+      df %>%
+        e_chart(Homework) %>%
+        e_bar(grade, barWidth = "50%") %>%
+        e_legend(show = F) %>%
+        e_y_axis(formatter = e_axis_formatter("percent", digits = 2)) %>%
+        e_labels(position = "top", formatter  = htmlwidgets::JS("
+                                    function(params){
+                                      return(parseFloat(params.value[1]*100).toFixed(2) +'%');
+                                    }
+                                    ")) %>%
+        e_grid(left = "15%", right = "5%", top = "10%", bottom = "10%")
+    })
+    
+    
+    output$homework_grades <- renderRHandsontable({
+      req(r$is$auth)
+      
+      df_hw <- df_filtered_homework_grades()
+      
+      # TODO: Conditional formatting for cells that are actually NA vs character of NA ----
+      rhandsontable(df_hw
+                    , rowHeaders = NULL
+                    , stretchH = 'all') 
+    })
+  }) #end module server
+}
+
+review_UI <- function(id) {
+  tabPanel(title = "Reviews"
+           , fluidRow(
+             column(width = 12
+              , box(width = 12, title = "Review Grades"
+                  , rHandsontableOutput(NS(id,"review_grades"))
+              )
+              , box(width = 12, title = "Topic Proficiency"
+                    , rHandsontableOutput(NS(id,"topic_proficiency"))
```

Comment:
Seems like math is off for remaining attempts?

![Screen Shot 2020-10-18 at 10 35 31 PM (2)](https://user-images.githubusercontent.com/52048606/96395559-f0eea080-1192-11eb-83f0-c8b4f2f24c07.png)

Reply:
The remaining attempts = total attempts - previous attempts. The issue derives from there being too many attempts because the review_to_topic table does not reflect what is in the review_topic table. For example, the review_topic table says only review 1 has topic 1, but in review_to_topic, reviews 4 and 1 have grades for topic 1, making the remaining attempts = -1.